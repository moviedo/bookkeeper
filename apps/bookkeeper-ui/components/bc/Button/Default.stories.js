export default {
  title: "base components / bc-button / default",
};

const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes).filter((a) => a !== "default"),
  template: `
  <bc-button v-bind="$props">
    ${args.default}
  </bc-button>
  `,
});

export const Default = Template.bind({});
Default.args = {
  default: "Submit",
  disabled: false,
  outline: false,
  small: false,
  large: false,
};

export const Tablet = Template.bind({});
Tablet.args = { ...Default.args };
Tablet.parameters = {
  viewport: {
    defaultViewport: "tablet",
  },
};

export const Mobile = Template.bind({});
Mobile.args = { ...Default.args };
Mobile.parameters = {
  viewport: {
    defaultViewport: "mobile1",
  },
};

export const WithCustomHtml = Template.bind({});
WithCustomHtml.args = {
  ...Default.args,
  default: `<i>${Default.args.default}</i> normal copy`,
};

export const Small = Template.bind({});
Small.args = {
  ...Default.args,
  small: true,
};

export const Large = Template.bind({});
Large.args = {
  ...Default.args,
  large: true,
};
