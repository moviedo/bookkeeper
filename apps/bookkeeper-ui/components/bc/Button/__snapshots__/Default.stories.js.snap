// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base components / bc-button / default Default smoke-test 2`] = `
<button data-v-85cb8f92
        class="default hover focus active"
>
  Submit
</button>
`;

exports[`base components / bc-button / default Large smoke-test 2`] = `
<button data-v-85cb8f92
        class="default hover focus active"
        large="true"
>
  Submit
</button>
`;

exports[`base components / bc-button / default Mobile smoke-test 2`] = `
<button data-v-85cb8f92
        class="default hover focus active"
>
  Submit
</button>
`;

exports[`base components / bc-button / default Small smoke-test 2`] = `
<button data-v-85cb8f92
        class="default hover focus active"
        small="true"
>
  Submit
</button>
`;

exports[`base components / bc-button / default Tablet smoke-test 2`] = `
<button data-v-85cb8f92
        class="default hover focus active"
>
  Submit
</button>
`;

exports[`base components / bc-button / default WithCustomHtml smoke-test 2`] = `
<button data-v-85cb8f92
        class="default hover focus active"
>
  <i data-v-85cb8f92>
    Submit
  </i>
  normal copy
</button>
`;
